gateserver/controller_protocol.py

```diff
@@ -0,0 +1,83 @@
+"""Implements the Controller <-> Server protocol.
+
+See https://github.com/fmfi-svt-gate/server/wiki/Controller-%E2%86%94-Server-Protocol .
+"""
+
+from .utils.structparse import *
+import nacl.raw as nacl
+
+PROTOCOL_VERSION = bytes([0x00,0x01])
+
+class BadMessageError(Exception): pass
+
+def checkmsg(expression, errmsg):
+    if not expression: raise BadMessageError(errmsg)
+
+class MsgType(Enum):
+    OPEN = 1
+
+class ReplyStatus(Enum):
+    OK        = 0x01
+    ERR       = 0x10
+    TRY_AGAIN = 0x11
+S = ReplyStatus
+
+PacketHead = mystruct('PacketHead',
+                     ['protocol_version', 'controllerID', 'nonce'      ],
+                     [ t.bytes(2)       ,  t.bytes(6)   ,  t.bytes(18) ])
```

Comment:
Úplne súhlasím. Ale ak vertikálne, tak to buď musia byť tuple, čo je hrozne veľa zátvoriek, alebo hlúposť typu "párne sú to, nepárne sú hento", čo sa mi hrozne nepáči... Ak vymyslíš nehrozné riešenie, poteším sa :D